### PR TITLE
Joint dislocation

### PR DESCRIPTION
**Pull request by MaxIsJoe**
Purpose
part of #6272
Adds in Joint dislocation for blunt force trauma damage.
Removes damage thresholds and only causes traumas to trigger upon reaching 100 damage.

Changelog:
CL : [New] - Joint Dislocation has been added.
CL : [Balance] - Traumas occur on near-death damage levels only now.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
pls review

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs [hunk 1]
```diff
@@ -226,6 +226,8 @@ public void AffectDamage(float damage, int damageType)
 			{
 				DamageOrgans(damage, attackType, damageType, organDamageSplit, armorPenetration);
 			}
+
+			if(damage < 10) return; //Do not apply traumas if the damage is not serious.
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: magic number alert!
Make it its own variable and expose it to editor as a SerializedField, you can use 10 as default value if  you wish.

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamageDiscriptors.cs [hunk 2]
```diff
@@ -114,6 +114,43 @@ public string GetFullBodyPartDamageDescReport()
 			return TranslateTags(report.ToString());
 		}
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamageDiscriptors.cs:116]: correct the name of the file and class, ``BodyPartDamageDescriptors``

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs [hunk 3]
```diff
@@ -79,11 +71,10 @@ public float CurrentInternalBleedingDamage
...
 		/// <summary>
 		/// Applies trauma damage to the body part, checks if it has enough protective armor to cancel the trauma damage
-		/// and automatically checks how big is the body part's cut size.
 		/// </summary>
 		public void ApplyTraumaDamage(TraumaticDamageTypes damageType = TraumaticDamageTypes.SLASH, bool ignoreSeverityCheck = false)
 		{
-			if(Severity < DamageSeverity.LightModerate && ignoreSeverityCheck == false) return;
+			if(Severity < DamageSeverity.Bad && ignoreSeverityCheck == false) return;
 			//We use dismember protection chance because it's the most logical value.
 			if(DMMath.Prob(SelfArmor.DismembermentProtectionChance * 100) == false)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs]: why is this multiplying by 100? Is ``DismembermentProtectionChance`` in decimal format?

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs [hunk 4]
```diff
@@ -96,23 +87,24 @@ public void ApplyTraumaDamage(TraumaticDamageTypes damageType = TraumaticDamageT
...
-				}
+				TakeBluntDamage();
 			}
 		}
 
 		public void TakeBluntDamage()
 		{
 			void TakeBluntLogic(BodyPart bodyPart)
 			{
+				if (bodyPart.currentBluntDamageLevel == TraumaDamageLevel.SMALL && DMMath.Prob(50))
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs]: another magic number

**Comment by corp-0** [comment]
BodyPartTraumaDamage is getting quite big

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamage.cs]: solved

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs (hunk 3 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs]: Aran probably forgot to remove that when he updated this variable

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartTraumaDamage.cs]: fixed

File: UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamageDiscriptors.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Damage/BodyPartDamageDiscriptors.cs:116]: done

**Review by NoooneyDude — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**